pkg/reconciler/controller.go

```diff
@@ -19,33 +20,34 @@ import (
 	"knative.dev/pkg/controller"
 	"knative.dev/pkg/kmeta"
 	"knative.dev/pkg/logging"
+	"knative.dev/pkg/system"
 )
 
 func NewController() func(context.Context, configmap.Watcher) *controller.Impl {
 	return func(ctx context.Context, _ configmap.Watcher) *controller.Impl {
+		ns := system.Namespace()
+		ctx = info.StoreNS(ctx, ns)
+
 		log := logging.FromContext(ctx)
 		run := params.New()
-		err := run.Clients.NewClients(ctx, &run.Info)
+		rinfo := &run.Info
+		rinfo.Controller = info.GetControllerInfoFromEnvOrDefault()
+		err := run.Clients.NewClients(ctx, rinfo)
 		if err != nil {
 			log.Fatal("failed to init clients : ", err)
 		}
+		ctx = info.StoreInfo(ctx, ns, &run.Info)
 
 		kinteract, err := kubeinteraction.NewKubernetesInteraction(run)
 		if err != nil {
 			log.Fatal("failed to init kinit client : ", err)
 		}
 
-		c := make(chan struct{})
-		go func() {
-			c <- struct{}{}
-			if err := run.WatchConfigMapChanges(ctx); err != nil {
-				log.Fatal("error from WatchConfigMapChanges from watcher reconciler : ", err)
-			}
-		}()
-		<-c
+		if err := run.UpdatePACInfo(ctx); err != nil {
```

Comment:
with this any new changes to configmap, reconciler will not pick up them

Reply:
yeah... I don't mind loosing that functionality until we get viper in....

Reply:
I changed it to keep the old behaviour (reading the configmap on every request/reconcilation)...